api/specs.proto

```diff
@@ -255,3 +255,18 @@ message ClusterSpec {
 	// Existing tasks will continue to use the previous default until rescheduled.
 	Driver default_log_driver = 7;
 }
+
+// ExecutorAttachmentSpec specifies the allocation information
+// for the executor attachment.
+message ExecutorAttachmentSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+
+	// NodeID specifies the node on which the attachment is to be placed
+	string node_id = 2 [(gogoproto.customname) = "NodeID"];
```

Comment:
Yes, originally I was getting the nodeid from `ca.RemoteNode(stream.Context())` but moved to this becasue it did not work on the worker, it worked only on the manager, and my goal was to get this work e2e first.
I am happy to derive the nodeID from the connection ctx if it works.